tests/integration/apigateway/test_apigateway_api.py

```diff
@@ -1,16 +1,257 @@
+import logging
 import os
+import time
+from operator import itemgetter
 
+import pytest
+from botocore.exceptions import ClientError
+
+from localstack.services.apigateway.helpers import TAG_KEY_CUSTOM_ID
+from localstack.testing.aws.util import is_aws_cloud
 from localstack.utils.files import load_file
+from localstack.utils.strings import short_uid
+
+LOG = logging.getLogger(__name__)
 
 # parent directory of this file
 PARENT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 OPENAPI_SPEC_PULUMI_JSON = os.path.join(PARENT_DIR, "files", "openapi.spec.pulumi.json")
 
 
+@pytest.fixture(autouse=True)
+def apigw_snapshot_transformer(snapshot):
+    snapshot.add_transformer(snapshot.transform.apigateway_api())
+
+
+@pytest.fixture(scope="class", autouse=True)
+def apigw_cleanup_before_run(apigateway_client):
+    # TODO: remove this once all tests are properly cleaning up and using fixtures
+    rest_apis = apigateway_client.get_rest_apis()
+    for rest_api in rest_apis["items"]:
+        delete_rest_api_retry(apigateway_client, rest_api["id"])
+
+
+def delete_rest_api_retry(client, rest_api_id: str):
+    try:
+        if is_aws_cloud():
+            # This is ugly but API GW returns 429 very quickly, and we want to be sure to clean up properly
+            cleaned = False
+            while not cleaned:
+                try:
+                    client.delete_rest_api(restApiId=rest_api_id)
+                    cleaned = True
+                except ClientError as e:
+                    error_message = str(e)
+                    if "TooManyRequestsException" in error_message:
+                        time.sleep(10)
+                    elif "NotFoundException" in error_message:
+                        break
+                    else:
+                        raise
+        else:
+            client.delete_rest_api(restApiId=rest_api_id)
+
+    except Exception as e:
+        LOG.debug("Error cleaning up rest API: %s, %s", rest_api_id, e)
+
+
+@pytest.fixture
+def apigw_create_rest_api(apigateway_client):
```

Comment:
Rest APIs are also tested on the `ext` project, we can duplicate this fixture or make it available outside the tests so it can reused. I'm not dogmatic, a bit of repetition sometimes is better than a dependency, but something to think about it.

Reply:
Agreed! In the next iteration for a different resource, I'll move all the fixtures to `conftest.py` so they're accessible everywhere. Thanks for pointing it out! 👍